.github/workflows/npmpublish.yml

```diff
@@ -0,0 +1,105 @@
+# This workflow will run tests using node and then publish a package to GitHub Packages when a release is created
+# For more information see: https://help.github.com/actions/language-and-framework-guides/publishing-nodejs-packages
+
+name: Node.js Package
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js
+      uses: actions/setup-node@v1
+    - name: Setup xvfb
+      run: |
+        Xvfb :99 -screen 0 1024x768x24 &
+        echo "::set-env name=DISPLAY:::99.0"
+    - name: Test
+      run: |
+        npm ci
+        npm test
+
+  publish-npm:
+    needs: test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 12
+          registry-url: https://registry.npmjs.org/
+      # Since we created the release in the UI, we need to find it.
+      # This step gets the release from the GITHUB_REF env var
+      - name: Get release
+        id: get_release
+        uses: bruceadams/get-release@v1.2.2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Setup and build
+        run: |
+          npm ci
+          npm install -g json
+          json -I -f package.json -e "this.version=\"$GITHUB_REF\""
+          json -I -f package-lock.json -e "this.version=\"$GITHUB_REF\""
```

Comment:
I'm torn as well. Having the version in the code is quite nice, but I'm not sure that we'd want to commit again since the tag would be created when the release is created in the UI. Maybe the flow should be very different and we start with a different action.

Reply:
On approach could be a [manually triggered](https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#manual-events) workflow:

- Version as input
- tests
- update package.json and commit
- build and pack
- update draft release with asset
- push docs
- publish to npm

anyway, most of the needed steps are here already, I'm ok with committing this and testing it out.

Reply:
Ok, perfect. I will commit soon and we can figure out how to test. We have some new features, so we can try to get a beta.4 out the door